### PR TITLE
migrate push_gradients method

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: elasticdl/python/common/tensor_utils.py [hunk 1]
```diff
@@ -97,15 +100,17 @@ def pb_to_indexed_slices(pb):
...
-        raise ValueError(
-            "IndexedSlices pb only accepts indices with one dimension, got %d",
-            len(slices.indices.shape),
-        )
-    pb.ids.extend(slices.indices)
+    indices = slices.indices
+    if isinstance(indices, np.ndarray):
+        if len(indices.shape) > 1:
+            raise ValueError(
+                "IndexedSlices pb only accepts indices with one"
```
Comment (workingloong) [inline elasticdl/python/common/tensor_utils.py]: Add a white space at the end of the line. Otherwise, the output string will be "indices with onedimension" not "indices with one dimension".

File: elasticdl/python/worker/ps_client.py [hunk 2]
```diff
@@ -92,3 +100,102 @@ def push_dense_parameters(self, parameters, ps_id, version):
...
             serialize_ndarray(p.values, model.dense_parameters[p.name])
         self.ps_stubs[ps_id].push_model(model)
+
+    def push_gradients(
+        self, grads, edl_grads, learning_rate, model_versions,
+    ):
+        """
+        Push gradients to PS. There three kinds of gradients:
+         - gradients
+         - sparse gradients of ElasticDL embedding layers
```
Comment (workingloong) [inline elasticdl/python/worker/ps_client.py:110]: The docstring only displays 2 kinds of gradients.

File: elasticdl/python/common/tensor_utils.py (hunk 1 above)
Comment (QiJune) [reply elasticdl/python/common/tensor_utils.py]: Done

File: elasticdl/python/worker/ps_client.py (hunk 2 above)
Comment (QiJune) [reply elasticdl/python/worker/ps_client.py:110]: Done

File: elasticdl/python/worker/ps_client.py [hunk 3]
```diff
@@ -92,3 +100,102 @@ def push_dense_parameters(self, parameters, ps_id, version):
         for p in parameters:
             serialize_ndarray(p.values, model.dense_parameters[p.name])
         self.ps_stubs[ps_id].push_model(model)
+
+    def push_gradients(
+        self, grads, edl_grads, learning_rate, model_versions,
+    ):
+        """
+        Push gradients to PS. There two kinds of gradients:
+         - gradients of normal layers
```
Comment (Yancey0623) [inline elasticdl/python/worker/ps_client.py:109]: Dense gradients of normal layers.

**Review by Yancey0623 — APPROVED** [review]
LGTM

File: elasticdl/python/worker/ps_client.py (hunk 3 above)
Comment (QiJune) [reply elasticdl/python/worker/ps_client.py:109]: Will fix it in next PRs

Comment (QiJune) [reply elasticdl/python/worker/ps_client.py:109]: Normal embedding layers have sparse gradients, while normal non-embedding layers have dense gradients.
